### PR TITLE
Add camlidl 1.07 package

### PR DESCRIPTION
**Pull request by yakobowski**

### CODE REVIEW

File: packages/camlidl/camlidl.1.07/files/disable-fatal-warn-31.diff [hunk 1]
```diff
@@ -0,0 +1,24 @@
+diff -Naur camlidl-1.05/config/Makefile.unix camlidl-1.05-patched/config/Makefile.unix
+--- camlidl-1.05/config/Makefile.unix	2002-04-22 12:50:46.000000000 +0100
++++ camlidl-1.05-patched/config/Makefile.unix	2016-05-04 13:10:23.370704022 +0100
+@@ -37,7 +37,7 @@
+ BINDIR=/usr/local/bin
+ 
+ # The Objective Caml compilers (the defaults below should be OK)
+-OCAMLC=ocamlc -g
++OCAMLC=ocamlc -g -warn-error -31
```
Comment (kit-ty-kate) [inline packages/camlidl/camlidl.1.07/files/disable-fatal-warn-31.diff]: This patch is called `disabled-fatal-warn` but enables a fatal warning. Fatal warnings should not be part of any releases

File: packages/camlidl/camlidl.1.07/opam [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+bug-reports: "https://github.com/xavierleroy/camlidl/issues"
+license: "QPL-1.0 and LGPL-2 with exceptions"
+build: [
+  ["mv" "config/Makefile.unix" "config/Makefile"] {os != "win32"}
+  ["mv" "config/Makefile.mingw" "config/Makefile"] {os = "win32"}
+  [make "all"]
+]
+patches: [
+  "disable-fatal-warn-31.diff" {ocaml:version >= "4.03"}
+  "camlidl-camlidl106.patch" {os = "win32"}
```
Comment (kit-ty-kate) [inline packages/camlidl/camlidl.1.07/opam]:
```suggestion
```
This patch doesn't exist

Comment (yakobowski) [reply packages/camlidl/camlidl.1.07/opam]: Right, sorry. I forgot to remove it from this section.

File: packages/camlidl/camlidl.1.07/files/disable-fatal-warn-31.diff (hunk 1 above)
Comment (kit-ty-kate) [reply packages/camlidl/camlidl.1.07/files/disable-fatal-warn-31.diff]: oh right, my bad

File: packages/camlidl/camlidl.1.07/opam (hunk 2 above)
Comment (yakobowski) [reply packages/camlidl/camlidl.1.07/opam]: Fixed.

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**